velox/common/caching/SimpleLRUCache.h

```diff
@@ -18,271 +18,145 @@
+        pinnedSize{curSize} {}
+
   // Capacity of the cache.
-  size_t maxSize{0};
+  const size_t maxSize;
 
   // Current cache size used.
-  size_t curSize{0};
-
-  // Current cache size used by pinned entries.
-  size_t pinnedSize{0};
-
-  // Total number of elements in the cache.
-  size_t numElements{0};
+  const size_t curSize;
 
   // Total number of cache hits since server start.
-  size_t numHits{0};
+  const size_t numHits;
 
   // Total number of cache lookups since server start.
-  size_t numLookups{0};
+  const size_t numLookups;
+
+  // These 2 are unused, but open source Presto depends on them
+  // Remove the usage in open source presto and get rid of them.
+  const size_t numElements;
+  const size_t pinnedSize;
 
   std::string toString() const {
     return fmt::format(
         "{{\n"
         "  maxSize: {}\n"
         "  curSize: {}\n"
-        "  pinnedSize: {}\n"
-        "  numElements: {}\n"
         "  numHits: {}\n"
         "  numLookups: {}\n"
         "}}\n",
         maxSize,
         curSize,
-        pinnedSize,
-        numElements,
         numHits,
         numLookups);
   }
+
+  bool operator==(const SimpleLRUCacheStats& rhs) const {
+    return this->curSize == rhs.curSize && this->maxSize == rhs.maxSize &&
+        this->numHits == rhs.numHits && this->numLookups == rhs.numLookups;
+  }
 };
 
-/// A simple LRU cache that allows each element to occupy an arbitrary
-/// amount of space in the cache. Useful when your the size of the
-/// cached elements can vary a lot; if they are all roughly the same
-/// size something that only tracks the number of elements in the
-/// cache like common/datastruct/LRUCacheMap.h may be better.
+/// A simple wrapper on top of the folly::EvictingCacheMap that tracks
+/// hit/miss counters. Key/Value evicted are immediately destructed.
+/// So the Key/Value should be a value type or self managing lifecycle
+/// shared_ptr.
 ///
 /// NOTE:
 /// 1. NOT Thread-Safe: All the public calls modify internal structures
 /// and hence require external write locks if used from multiple threads.
-/// 2. 'Key' is required to be copyable and movable.
-template <
-    typename Key,
-    typename Value,
-    typename Comparator = std::equal_to<Key>,
-    typename Hash = std::hash<Key>>
+template <typename Key, typename Value>
 class SimpleLRUCache {
  public:
-  /// Constructs a cache of the specified size. This size can represent whatever
-  /// you want -- slots, or bytes, or etc; you provide the size of each element
-  /// whenever you add a new value to the cache. Note that in certain
-  /// circumstances this max_size may be exceeded -- see add().
-  explicit SimpleLRUCache(size_t maxSize);
-
-  /// Frees all owned data. Check-fails if any element remains pinned.
-  ~SimpleLRUCache();
+  /// Constructs a cache of the specified size. The size represents the number
+  /// of entries in the cache.
+  explicit SimpleLRUCache(size_t maxSize, size_t clearSize = 1);
```

Comment:
Can you comment clearSize which is used to configured the number of items to evict each time in EvictingCacheMap? Thanks!